compiler/rustc_trait_selection/src/traits/fulfill.rs

```diff
@@ -473,6 +473,25 @@ impl<'a, 'b, 'tcx> ObligationProcessor for FulfillProcessor<'a, 'b, 'tcx> {
 
                 ty::PredicateAtom::ConstEquate(c1, c2) => {
                     debug!("equating consts: c1={:?} c2={:?}", c1, c2);
+                    if self.selcx.tcx().features().const_evaluatable_checked {
+                        // FIXME: we probably should only try to unify abstract constants
+                        // if the constants depend on generic parameters.
+                        //
+                        // Let's just see where this breaks :shrug:
```

Comment:
maybe make the `try_unify_abstract_consts` query a `raw` query and add a wrapper that has a fast path for such checks